### PR TITLE
[ADD] payment_providers: Xendit

### PR DESCRIPTION
**Pull request by nni-odoo**
Documentation for Xendit payment provider

taskid-2946329

See also:
- https://github.com/odoo/odoo/pull/141661

### CODE REVIEW

**Comment by nni-odoo** [comment]
Hi @toaa-odoo this is the documentation for the master branch. I've added you as the reviewer here since you reviewed my PR before (to 16.3) https://github.com/odoo/documentation/pull/5852/commits. Thanks again!

**Review by toaa-odoo — APPROVED** [review]
Tagging the next reviewer in line :)

**Comment by xpl-odoo** [comment]
Hi @nni-odoo ! I pushed some changes to go faster.

I removed the part about checking the boxes below the "Invoices: Invoices paid" URL in the Webhook section. This didn't feel necessary (you _can_) without explaining what's their use. But feel free to add it back if you believe it's essential.

You can r+ whenever.

@robodoo delegate+

File: content/applications/finance/payment_providers/xendit.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+
+Configuration on Odoo
+=====================
+
+#. :ref:`Navigate to the payment provider Xendit <payment_providers/add_new>` and change its state
+   to :guilabel:`Enabled`.
+#. Fill in the :guilabel:`Xendit API Key` and :guilabel:`Xendit Callback Token` fields with the
+   information saved from the previous steps.
+#. Configure the payment methods you want to activate on the configuration tab.
```
Comment (auva-odoo) [inline content/applications/finance/payment_providers/xendit.rst:52]:
```suggestion
#. Configure the rest of the options to your liking.
```
since it's not just the payment methods :)

File: content/applications/finance/payment_providers/xendit.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+   - :guilabel:`Invoices: Invoices paid`
+
+
+Configuration on Odoo
+=====================
+
+#. :ref:`Navigate to the payment provider Xendit <payment_providers/add_new>` and change its state
+   to :guilabel:`Enabled`.
+#. Fill in the :guilabel:`Xendit API Key` and :guilabel:`Xendit Callback Token` fields with the
```
Comment (auva-odoo) [inline content/applications/finance/payment_providers/xendit.rst:50]: From what I can see in the dev PR, but feel free to correct this if I'm wrong:

```suggestion
#. Fill in the :guilabel:`Secret Key` and :guilabel:`Webhook Token` fields with the
```

File: content/applications/finance/payment_providers/xendit.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+   information saved from the previous steps.
```
Comment (auva-odoo) [inline content/applications/finance/payment_providers/xendit.rst:51]: I would add a ref here linking to the first h2 (xendit dashboard config), like we have for other payment providers

**Review by auva-odoo — CHANGES_REQUESTED** [review]
Hi @nni-odoo thank you very much for writing this doc! 
In addition to my comments, I'd like to suggest merging all sub-sections under the first h2 to have just one big numbered list with all the steps that need to be done on the Xendit dashboard. 
If it's ok with you, don't forget to remove repetitions, e.g., we don't need to go to Configuration: Settings and select Developers a second time, we can just stay on the same page and scroll down to the Webhooks section (or select it in the small menu) and go from there:
xendit-developers-menu

Don't hesitate if you have any questions!

**Comment by nni-odoo** [comment]
@auva-odoo thank you for the suggestions! I've revised the document and followed your suggestion on combining the sub-sections under the first h2. Let me know if there's anything else I should revise!

File: content/applications/finance/payment_providers/xendit.rst [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+=====================================
+
+#. Create a Xendit account if necessary and log in to the `Xendit Dashboard
+   <https://dashboard.xendit.co>`_.
+#. Check your account mode in the top left corner of the page. Use the :guilabel:`Test Mode` to try the integration
+   without charging your customers. Switch to :guilabel:`Live Mode` once you are ready to accept
+   payments.
+#. Navigate to :menuselection:`Configuration: Settings` in the left part of the application page. In the :guilabel:`Developers` section,
+   click `API Keys <https://dashboard.xendit.co/settings/developers#api-keys>`_. In tis page, you will setup for both :guilabel:`API key`
+   and :guilabel:`webhook token`.
```
Comment (auva-odoo) [inline content/applications/finance/payment_providers/xendit.rst:21]: I don't think the second part of the sentence is necessary. I think it's fine if users follow the rest of the procedure and find out what they need to when they get to it?
If you really want to keep it, then I would change it to:

This page allows you to set up both the API key and webhook token. 
(no guilabel, but you could use bold if you want to)

File: content/applications/finance/payment_providers/xendit.rst [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+#. Click :guilabel:`Generate Secret Key`, and in the popup box, enter any :guilabel:`API key name`, select :guilabel:`Write` for
+   the :guilabel:`Money-in Products` permission and :guilabel:`None` for all other permissions
+   to keep your data secured, then click :guilabel:`Generate key`.
```
Comment (auva-odoo) [inline content/applications/finance/payment_providers/xendit.rst:24]: This sentence seems a little long so I would rephrase it as follows:
```suggestion
#. Click :guilabel:`Generate Secret Key`. In the popup box, enter an :guilabel:`API key name`, select :guilabel:`Write` for  the :guilabel:`Money-in Products` permission and :guilabel:`None` for all other permissions, then click :guilabel:`Generate key`.
```

File: content/applications/finance/payment_providers/xendit.rst [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+#. Confirm your password to display your API key. Copy or download the key and **save
+   this information securely for later**. This is the only time the API key can be viewed or downloaded.
+#. When completed, scroll down the page to the `Webhooks <https://dashboard.xendit.co/settings/developers#webhooks>`_ section.
```
Comment (auva-odoo) [inline content/applications/finance/payment_providers/xendit.rst:27]:
```suggestion
#. Once completed, scroll down the page to the `Webhooks <https://dashboard.xendit.co/settings/developers#webhooks>`_ section to generate the webhook token.
```
(added the last part since i removed it from the next bullet point and i thought it would work well here)

File: content/applications/finance/payment_providers/xendit.rst [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+#. Under :guilabel:`Webhook verification token`, click :guilabel:`View Webhook Verification Token` to generate the token.
+#. Confirm your password to display the token and save it for later.
```
Comment (auva-odoo) [inline content/applications/finance/payment_providers/xendit.rst:29]: I think these two can be merged:
```suggestion
#. Under :guilabel:`Webhook verification token`, click :guilabel:`View Webhook Verification Token`, then confirm your password to display the token. Save it for later.
```

File: content/applications/finance/payment_providers/xendit.rst [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+   :guilabel:`Test and save` button next to it.
+
+
+Configuration on Odoo
+=====================
+
+#. :ref:`Navigate to the payment provider Xendit <payment_providers/add_new>` and change its state
+   to :guilabel:`Enabled`.
+#. Fill in the :guilabel:`Secret Key` and :guilabel:`Webhook Token` fields with the
+   information saved from the previous steps :ref:`payment_providers/xendit/configure_dashboard`.
```
Comment (auva-odoo) [inline content/applications/finance/payment_providers/xendit.rst:41]:
```suggestion
#. Fill in the :guilabel:`Secret Key` and :guilabel:`Webhook Token` fields with the
   information saved at the step :ref:`payment_providers/xendit/configure_dashboard`.
```
(nitpicking, this is just to have the same as the other (recently updated) payment provider docs)

File: content/applications/finance/payment_providers/xendit.rst [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+   to keep your data secured, then click :guilabel:`Generate key`.
+#. Confirm your password to display your API key. Copy or download the key and **save
+   this information securely for later**. This is the only time the API key can be viewed or downloaded.
+#. When completed, scroll down the page to the `Webhooks <https://dashboard.xendit.co/settings/developers#webhooks>`_ section.
+#. Under :guilabel:`Webhook verification token`, click :guilabel:`View Webhook Verification Token` to generate the token.
+#. Confirm your password to display the token and save it for later.
+#. In the :guilabel:`Webhook URL` section, enter your Odoo database URL (e.g.,
+   `https://example.odoo.com`) in the field :guilabel:`Invoices paid` and click the
+   :guilabel:`Test and save` button next to it.
+
```
Comment (auva-odoo) [inline content/applications/finance/payment_providers/xendit.rst:33]: there a 2 empty lines here instead of just one

**Review by auva-odoo — CHANGES_REQUESTED** [review]
@nni-odoo I've added a few more suggestions/comments. Also, could you please keep lines within 100 characters? Thank you!

**Comment by nni-odoo** [comment]
@auva-odoo revised the doc again and limited each line to <= 100 characters! Let me know if there's anything else!

**Comment by nni-odoo** [comment]
@auva-odoo Thank you for your help! I'll wait for the main PR https://github.com/odoo/odoo/pull/141661 to be merged.
On that note, do I need to squash all commits when merging? If so, is it done automatically through r+?

**Comment by auva-odoo** [comment]
@nni-odoo Yes you do need to squash them before merging; it's note done automatically

**Comment by AntoineVDV** [comment]
@nni-odoo Could you squash the commits *without rebasing on a fresh 17.0* (use `git rebase -i HEAD~4`) and r+ yourself?

**Comment by nni-odoo** [comment]
@robodoo r+

**Comment by nni-odoo** [comment]
@AntoineVDV thank you!
